doc/development/guide/documentation.rst

```diff
@@ -1,39 +1,449 @@
+    --- of the documented code object.
 
 .. note::
 
-    To build the interfaces documentation, PyTorch and TensorFlow will need to
-    be installed, see :ref:`install_interfaces`.
+    Long-form tutorials, as well as quantum machine learning theory and background information,
+    is not part of the PennyLane documentation. Instead, this content is available over in the
+    `XanaduAI/QML <https://github.com/XanaduAI/qml>`_` GitHub repository (viewable at
+    https://pennylane.ai/qml).
+
+.. _docstrings:
+
+Docstrings
+----------
+
+Documentation strings (docstrings) describe the use and functionality of modules, functions,
+classes, and variables. Docstrings in PennyLane are enclosed within triple-quotes (``"""``), and are
+written using `reStructuredText (ReST) <https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html>`_
+markup, and are formatted `Google-style <https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html#google-vs-numpy>`_.
+
+Docstrings are a part of Python; an objects' docstring is stored under the ``__doc__``
+attribute, and accessible to the user via the ``help()`` function. In addition, docstrings
+are automatically extracted and rendered by Sphinx, and inserted into the
+:doc:`API </code/qml>` section of the documentation.
+
+Guiding principles
+~~~~~~~~~~~~~~~~~~
+
+Below, we list some general guidelines, before discussing required docstring styles and formatting
+for various PennyLane objects.
+
+* **Be clear and concise, and avoid redundant information.** Favour clarity over completeness; too
+  much information can distract from conveying the functionality of the code unit.
+  Assume the reader has basic domain knowledge, and provide links to internal/external resources
+  where deemed necessary.
+
+* **Be mindful with ReST markup**. Overly complex formatting can make the content difficult
+  to read for users viewing the docstring in a terminal/Jupyter environment. Try to also avoid
+  complex formatting such as tables, which can be difficult to maintain over time.
+
+* **Do not assume the reader can view the accompanying code.** The reader will most likely
+  be viewing the docstring via the Sphinx-rendered website, or in a terminal/Jupyter environment.
+  For documenting how the code itself works, favour comments.
+
+* **Cross reference where applicable.** Use the roles ``:class:``, ``:func:``, ``:attr:``,
+  ``:meth:``, and ``:mod:`` where applicable to have Sphinx automatically include a cross-reference
+  to the specified code object.
+
+  For example:
+
+  .. code-block:: rest
+
+      The recommended method of creating a QNodeCollection is via :func:`~.map`.
+
+  Here, ``~`` instructs Sphinx to display just the object name, as opposed to the full absolute path
+  of the object (``pennylane.map()``). To use custom text, you may use the syntax
+  ``:func:`map function <~.map>```.
+
+* **Favour document links over reference links.** If linking to a particular page, use
```

Comment:
I wouldn't even say "favour". If the `:doc:` option is available, we should use it!